File: CSharpCodeBase/components/thing/fragilethingcomponent.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class fragilethingcomponent :component{
 public void init(self, entity);
       Component.init(self, entity);
 }})
 public void OnCollisionEnter(targetEntity){
   if(targetEntity.characterClass != null  &&  targetEntity.characterClass > 1  ){ 
     this.entity:OnFragile(targetEntity);
   }
 }
 }}

File: CSharpCodeBase/components/thing/fragilethingcomponent.lua
local Class = require 'utils.hump.class'
local Component = require 'components.component'

local FragileThingComponent = Class({__includes=Component, init=function(self, entity)
      Component.init(self, entity)
end})

function FragileThingComponent:OnCollisionEnter(targetEntity)
  if targetEntity.characterClass ~= nil and targetEntity.characterClass > 1 then
    self.entity:OnFragile(targetEntity)
  end
end

return FragileThingComponent